File: bit-manipulation/number-of-steps-to-reduce-a-number-to-zero/number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
function numberOfSteps(num) {
  if (num == 0) return 0;

  let count = 0;

  while (num > 0) {
    if ((num & 1) === 1) {
      count += 2;
    } else {
      count++;
    }
    num >>= 1;
  }
  return count - 1;
}

File: bit-manipulation/number-of-steps-to-reduce-a-number-to-zero/number-of-steps-to-reduce-a-number-to-zero.go
func numberOfSteps(num int) int {
	if num == 0 {
		return 0
	}

	count := 0

	for num > 0 {
		if (num & 1) == 1 {
			count += 2
		} else {
			count++
		}
		num >>= 1
	}
	return count - 1
}
